connectors/embeddings-vector-database/src/main/java/io/camunda/connector/embeddingstore/DefaultEmbeddingStoreFactory.java

```diff
@@ -39,6 +60,97 @@ public EmbeddingStore<TextSegment> initializeVectorStore(
+
+    final var cosmosClientBuilder = new CosmosClientBuilder();
+    cosmosClientBuilder.endpoint(azureCosmosDbNoSqlVectorStore.endpoint());
+    cosmosClientBuilder.consistencyLevel(
+        mapConsistencyLevel(azureCosmosDbNoSqlVectorStore.consistencyLevel()));
+    cosmosClientBuilder.contentResponseOnWriteEnabled(true);
+
+    switch (azureCosmosDbNoSqlVectorStore.authentication()) {
+      case AzureCosmosDbNoSqlVectorStore.AzureAuthentication.AzureApiKeyAuthentication apiKey ->
+          cosmosClientBuilder.key(apiKey.apiKey());
+      case AzureCosmosDbNoSqlVectorStore.AzureAuthentication.AzureClientCredentialsAuthentication
+              auth -> {
+        ClientSecretCredentialBuilder clientSecretCredentialBuilder =
+            new ClientSecretCredentialBuilder()
+                .clientId(auth.clientId())
+                .clientSecret(auth.clientSecret())
+                .tenantId(auth.tenantId());
+        if (StringUtils.isNotBlank(auth.authorityHost())) {
+          clientSecretCredentialBuilder.authorityHost(auth.authorityHost());
+        }
+        cosmosClientBuilder.credential(clientSecretCredentialBuilder.build());
+      }
+    }
+
+    final var embedding = new CosmosVectorEmbedding();
+    embedding.setPath(AZURE_COSMOS_DB_VECTOR_EMBEDDING_PATH);
+    embedding.setDataType(CosmosVectorDataType.FLOAT32);
+    embedding.setEmbeddingDimensions(model.dimension());
+    embedding.setDistanceFunction(
+        mapDistanceFunction(azureCosmosDbNoSqlVectorStore.distanceFunction()));
+    final var embeddingPolicy = new CosmosVectorEmbeddingPolicy();
+    embeddingPolicy.setCosmosVectorEmbeddings(List.of(embedding));
+
+    var vectorIndexSpec = new CosmosVectorIndexSpec();
+    vectorIndexSpec.setPath(AZURE_COSMOS_DB_VECTOR_EMBEDDING_PATH);
+    vectorIndexSpec.setType(
+        mapIndexType(azureCosmosDbNoSqlVectorStore.vectorIndexType()).toString());
+
+    final var partitionKeyDef = new PartitionKeyDefinition();
+    partitionKeyDef.setPaths(List.of(AZURE_COSMOS_DB_PARTITION_KEY_PATH));
+
+    final var containerProperties =
+        new CosmosContainerProperties(
+            azureCosmosDbNoSqlVectorStore.containerName(), partitionKeyDef);
+    final var indexingPolicy = new IndexingPolicy();
+    indexingPolicy.setIndexingMode(IndexingMode.CONSISTENT);
+    indexingPolicy.setIncludedPaths(List.of(new IncludedPath("/*")));
+    containerProperties.setIndexingPolicy(indexingPolicy);
+
+    return AzureCosmosDbNoSqlEmbeddingStore.builder()
+        .cosmosClient(cosmosClientBuilder.buildClient())
+        .databaseName(azureCosmosDbNoSqlVectorStore.databaseName())
+        .containerName(azureCosmosDbNoSqlVectorStore.containerName())
+        .cosmosVectorEmbeddingPolicy(embeddingPolicy)
+        .cosmosVectorIndexes(List.of(vectorIndexSpec))
+        .containerProperties(containerProperties)
+        .build();
+  }
+
+  private ConsistencyLevel mapConsistencyLevel(
```

Comment:
💭 Could we maybe include `AzureCosmos` in the naming of these functions to indicate where they belong to? Alternatively, as the Cosmos DB setup seems to be rather involved WDYT of extracting that functionality into a `AzureCosmosDbEmbeddingStoreFactory` used from this one?

Reply:
I agree with extracting `AzureCosmosDbEmbeddingStoreFactory` as the file would eventually grow too long by adding more embedding stores.